tools/cldr-code/src/main/java/org/unicode/cldr/test/CoverageLevel2.java

```diff
@@ -179,38 +203,175 @@ public int getIntLevel(String path) {
+    // Moved code in from SupplementalInfo
+    //
+    // TODO:
+    // 1. drop the corresponding code in SupplementalInfo.
+    // 2. change SupplementalInfo to skip reading coverageLevels.xml
+    // 3. change the default creation of CoverageLevels2 to instead use this code with that file.
+    // Later
+    // 4. Generalize the RawCoverageFile code, and use with other supplemental files.
+    //    That way supplemental files can be read as needed instead of all at once.
+
+    final private List<String> approvalRequirements = new LinkedList<>(); // xpath array
+    private VariableReplacer coverageVariables = new VariableReplacer();
+    private SortedSet<CoverageLevelInfo> coverageLevels = new TreeSet<>();
+
+    public class RawCoverageFile {
+
+        private VersionInfo cldrVersion;
+
+        class MyHandler extends XMLFileReader.SimpleHandler {
+            @Override
+            public void handlePathValue(String path, String pathValue) {
+                XPathParts parts = XPathParts.getFrozenInstance(path);
+                String level1 = parts.size() < 2 ? null : parts.getElement(1);
+                if (level1.equals("version")) {
+                    if (cldrVersion == null) {
+                        String version = parts.getAttributeValue(1, "cldrVersion");
+                        if (version == null) {
+                            version = parts.getAttributeValue(0, "version");
+                        }
+                        cldrVersion = VersionInfo.getInstance(version);
+                    }
+                } else if (parts.containsElement("approvalRequirement")) {
+                    approvalRequirements.add(parts.toString());
+                } else if (parts.containsElement("coverageLevel")) {
+                    String match = parts.containsAttribute("match") ? coverageVariables.replace(parts.getAttributeValue(-1,
+                        "match")) : null;
+                    String valueStr = parts.getAttributeValue(-1, "value");
+                    // Ticket 7125: map the number to English. So switch from English to number for construction
+                    valueStr = Integer.toString(Level.get(valueStr).getLevel());
+
+                    String inLanguage = parts.containsAttribute("inLanguage") ? coverageVariables.replace(parts
+                        .getAttributeValue(-1, "inLanguage")) : null;
+                    String inScript = parts.containsAttribute("inScript") ? coverageVariables.replace(parts
+                        .getAttributeValue(-1, "inScript")) : null;
+                    String inTerritory = parts.containsAttribute("inTerritory") ? coverageVariables.replace(parts
+                        .getAttributeValue(-1, "inTerritory")) : null;
+                    Integer value = (valueStr != null) ? Integer.valueOf(valueStr) : Integer.valueOf("101");
+                    if (cldrVersion.getMajor() < 2) {
+                        value = 40;
```

Comment:
Not quite sure what the above 3 lines are trying to do, but I will not worry too much about it....

Reply:
Right